ext/ldap/ldap.c

```diff
@@ -3735,35 +3738,38 @@ int _ldap_rebind_proc(LDAP *ldap, const char *url, ber_tag_t req, ber_int_t msgi
 PHP_FUNCTION(ldap_set_rebind_proc)
 {
 	zval *link;
-	zend_fcall_info fci;
-	zend_fcall_info_cache fcc;
+	zend_fcall_info dummy_fci;
+	zend_fcall_info_cache fcc = empty_fcall_info_cache;
 	ldap_linkdata *ld;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "Of!", &link, ldap_link_ce, &fci, &fcc) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "OF!", &link, ldap_link_ce, &dummy_fci, &fcc) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	ld = Z_LDAP_LINK_P(link);
-	VERIFY_LDAP_LINK_CONNECTED(ld);
-
-	if (!ZEND_FCI_INITIALIZED(fci)) {
-		/* unregister rebind procedure */
-		if (!Z_ISUNDEF(ld->rebindproc)) {
-			zval_ptr_dtor(&ld->rebindproc);
-			ZVAL_UNDEF(&ld->rebindproc);
-			ldap_set_rebind_proc(ld->link, NULL, NULL);
-		}
-		RETURN_TRUE;
+	/* Inline VERIFY_LDAP_LINK_CONNECTED(ld); as we need to free trampoline */
+	if (!ld->link) {
+		zend_release_fcall_info_cache(&fcc);
+		zend_throw_error(NULL, "LDAP connection has already been closed");
+		RETURN_THROWS();
 	}
 
-	/* register rebind procedure */
-	if (Z_ISUNDEF(ld->rebindproc)) {
+	/* Free old FCC */
+	if (!ZEND_FCC_INITIALIZED(ld->rebind_proc_fcc)) {
+		zend_fcc_dtor(&ld->rebind_proc_fcc);
+		memcpy(&ld->rebind_proc_fcc, &empty_fcall_info_cache, sizeof(zend_fcall_info_cache));
```

Comment:
You don't need this copy, `zend_fcc_dtor` already does this for you.

Reply:
Sometimes I forget that I design APIs in a reasonable way